File: job_message_writer/backend/app/schemas/resume.py
# File: backend/app/schemas/resume.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class ResumeBase(BaseModel):
    title: str  # Profile title (e.g., "My Frontend Resume")

class ResumeCreate(ResumeBase):
    # File is handled by Form in the API endpoint, not here
    pass

class ResumeInfo(BaseModel):
    name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    skills: Optional[List[str]] = None
    years_experience: Optional[str] = None
    education: Optional[str] = None
    recent_job: Optional[str] = None
    recent_company: Optional[str] = None

class ProfileClassification(BaseModel):
    profile_type: Optional[str] = None  # Frontend, Backend, Full Stack, etc.
    primary_languages: Optional[List[str]] = None
    frameworks: Optional[List[str]] = None
    years_experience: Optional[str] = None
    seniority: Optional[str] = None
    industry_focus: Optional[str] = None

class ResumeResponse(BaseModel):
    id: int
    title: str
    filename: Optional[str] = None
    is_active: Optional[bool] = None
    extracted_info: Optional[ResumeInfo] = None
    profile_classification: Optional[ProfileClassification] = None
    
    class Config:
        orm_mode = True

class ResumeContentResponse(ResumeResponse):
    content: str  # The extracted text from the PDF

File: job_message_writer/frontend/src/api/resume.ts
import apiClient from './client';

// Types
export interface Resume {
  id: number;
  title: string;
  filename: string;
  is_active: boolean;
  extracted_info: {
    name: string;
    email: string;
    phone: string;
    skills: string[];
    years_experience: string;
    education: string;
    recent_job: string;
    recent_company: string;
  };
  profile_classification: {
    profile_type: string;
    primary_languages: string[];
    frameworks: string[];
    years_experience: string;
    seniority: string;
    industry_focus: string;
  };
}

// Upload a resume
export const uploadResume = async (file: File, title: string, makeActive: boolean = true) => {
  const formData = new FormData();
  formData.append('file', file);
  formData.append('title', title);
  formData.append('make_active', String(makeActive));
  
  const response = await apiClient.post<Resume>('/resumes/', formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  });
  
  return response.data;
};

// Get all resumes
export const getResumes = async () => {
  const response = await apiClient.get<Resume[]>('/resumes/');
  return response.data;
};

// Get active resume
export const getActiveResume = async () => {
  const response = await apiClient.get<Resume>('/resumes/active');
  return response.data;
};

// Set a resume as active
export const setActiveResume = async (resumeId: number) => {
  const response = await apiClient.post<Resume>(`/resumes/${resumeId}/set-active`);
  return response.data;
};

// Get resume by ID
export const getResume = async (resumeId: number) => {
  const response = await apiClient.get<Resume>(`/resumes/${resumeId}`);
  return response.data;
};

// Get resume content by ID
export const getResumeContent = async (resumeId: number) => {
  const response = await apiClient.get<Resume & { content: string }>(`/resumes/${resumeId}/content`);
  return response.data;
};